### PR TITLE
drivers: move disk drivers (and sdmmc controller drivers) to drivers/disk

### PR DESCRIPTION
**Pull request by jfischer-no**
The files disk_access_usdhc.c, disk_access_spi_sdhc.c, and
disk_access_stm32_sdmmc.c are actually drivers for SD/MMC
controllers. This patch moves this drivers to drivers/sdmmc and
reworks the configuration so that the drivers are selected when
the corresponding node is enabled.

### CODE REVIEW

**Comment by erwango** [comment]
^^ @anthonybrandon

**Comment by anthonybrandon** [comment]
I haven't tested on a board yet, but so far it looks fine.

**Comment by galak** [comment]
Dev-review (Feb 25): 
* split out driver related bits in include/disk/disk_access.h into include/drivers/disk.h
* Move drivers out of subsys/disk/{flash,ram,*sd*, *mmc*} into drivers/disk

**Review by carlescufi — APPROVED** [review]
OK on the move/rename as agreed during the dev-review meeting.

**Comment by carlescufi** [comment]
@MaureenHelm could you please review?

File: boards/arm/mimxrt1050_evk/pinmux.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ static gpio_pin_config_t enet_gpio_config = {
 };
 #endif
 
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(usdhc1), okay) && CONFIG_DISK_ACCESS_USDHC1
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(usdhc1), okay) && CONFIG_DISK_DRIVER_SDMMC
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply boards/arm/mimxrt1050_evk/pinmux.c:20]: No

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Looks good. Tested `samples/subsys/usb/mass` and `samples/subsys/fs/fat_fs` on `mimxrt1064_evk`.

Please also update the release notes

File: drivers/disk/Kconfig.sdmmc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_ZEPHYR_MMC_SPI_SLOT := zephyr,mmc-spi-slot
+DT_COMPAT_NXP_USDHC := nxp,imx-usdhc
+DT_COMPAT_ST_STM32_SDMMC := st,stm32-sdmmc
+
+config DISK_DRIVER_SDMMC
+	bool "SDMMC card driver"
+	select FLASH
```
[the remark this answers is not in the document]
Comment (jfischer-no) [reply drivers/disk/Kconfig.sdmmc:11]: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/disk/Kconfig#L97, but that is good question. These depends and selects around FLASH* overall in the tree annoy me a lot.

[the remark this answers is not in the document]
Comment (jfischer-no) [reply drivers/disk/Kconfig.sdmmc]: fixed

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**